datastore/query.go

```diff
@@ -536,9 +553,13 @@ func (q *Query) Run(c context.Context) *Iterator {
 		return t
 	}
 	offset := q.offset - t.res.GetSkippedResults()
+	count := t.limit
+	if t.count > 0 && (count == 0 || t.count < count) {
```

Comment:
I think assigning `t.limit` to local var `count` and then checking for it is a bit confusing here. 

Would not it be better to simply use `t.limit` in `if`?
```
	var count int
	if t.count > 0 && (t.limit < 0 || t.count < t.limit) {
		count = t.count
	} else {
		count = t.limit
	} 
```
@mtraver will appreciate if you can point out what is purpose allowing to pass `0` as `count`. It may result in unnecessary datastore request that returns nothing. Wouldn't it be more logical to to treat `0`s as unlimited in which case to not pass it over wire in `*pb.Query` and if a negative number passed to `Limit()` or `Count()` either return error or assign 0 internally? It's not obviously clear from current implementation what happens if you pass `count=0` to `internal.Call(c, "datastore_v3", "RunQuery", &req, &t.res)` - would it return 0 results or be treated as unlimited?

Reply:
I agree re: the local var. Your suggestion is more readable. Could also be

```
count := t.limit
if t.count > 0 && (t.limit < 0 || t.count < t.limit) {
	count = t.count
}
```

Yes, a `count` of 0 doesn't make sense, at least when passed in by the user. If a value <= 0 is passed to `BatchSize` then an error should be returned.

A `limit` of 0 is ok though, and it's actually used internally. See `Query`'s `Count` method, which includes this comment:

```
We also set the limit to zero, as we don't want any actual entity data,
just the number of skipped results.
```

Reply:
@mtraver thanks, `-1` and negative limit in query now makes sense.